xonsh/history/json.py

```diff
@@ -446,12 +446,7 @@ def __init__(
             if save_cwd is not None
             else XSH.env.get("XONSH_HISTORY_SAVE_CWD", True)
         )
-        ignore_regex = XSH.env.get("XONSH_HISTORY_IGNORE_REGEX")
-        self.is_ignored = (
-            (lambda cmd: re.match(ignore_regex, cmd["inp"]) is not None)
-            if ignore_regex is not None
-            else self.is_ignored
-        )
+        self.history_ignore_regex = XSH.env.get("XONSH_HISTORY_IGNORE_REGEX")
```

Comment:
I think it is better to move this to the base class as well. We don't need to repeat unless there is any specialization

Reply:
`History` is not currently aware of `XSH`. Is it OK to introduce this? I'm cool with it if you are. @anki-code I'm defending your honor, you got an opinion?

Do you want an intermediate class from which both the JSON and SQLite derived, and we leave History ignorant of `XSH`?